include/zephyr/kernel/internal/sched_priq.h

```diff
@@ -10,6 +10,18 @@
 #include <zephyr/sys/dlist.h>
 #include <zephyr/sys/rb.h>
 
+#define K_HIGHEST_THREAD_PRIO (-CONFIG_NUM_COOP_PRIORITIES)
+#define K_LOWEST_THREAD_PRIO CONFIG_NUM_PREEMPT_PRIORITIES
+#define K_IDLE_PRIO K_LOWEST_THREAD_PRIO
+#define K_HIGHEST_APPLICATION_THREAD_PRIO (K_HIGHEST_THREAD_PRIO)
+#define K_LOWEST_APPLICATION_THREAD_PRIO (K_LOWEST_THREAD_PRIO - 1)
+#define K_THRAD_PRIO_NUM (K_LOWEST_THREAD_PRIO - K_HIGHEST_THREAD_PRIO + 1)
```

Comment:
Spelling.  Also not really a fan of adding this as an API, maybe move this to the C file and call it "NUM_PRIO" or something simpler?